aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -181,14 +169,21 @@ void prepare_matrices_for_broadcasting(
       });
       cachedGraph = static_cast<CachedGraph *>(tmpCachedGraph);
     }
-    Placeholder selfPlaceholder = Placeholder(cachedGraph->selfTensor_, self);
-    Placeholder otherPlaceholder = Placeholder(cachedGraph->otherTensor_, other);
+    Placeholder selfPlaceholder = Placeholder();
```

Comment:
I am not following the logic here, when one of the `self` or `other` numel==0 and other one non-zero it will create empty placeholders. How does this logic handle that?

Reply:
Yeah I agree, its confusing to use output sizes to determine that. I was thinking it should be :
```
self.numel() == 0 || other.numel() == 0
```
That should cover all the cases. But I guess in that sense, its captured through output sizes. So lets leave it as is.